atdcpp/src/atdcpp_names.ml

```diff
@@ -0,0 +1,155 @@
+open Atd.Import
+(* Names *)
+
+let to_camel_case (s : string) =
+  let res    = Bytes.of_string s in
+  let offset = ref 0 in
+  let upper  = ref true in
+  let f = function
+    | '_' ->
+        upper := true;
+    | ('0'|'1'|'2'|'3'|'4'|'5'|'6'|'7'|'8'|'9') as x ->
+        upper := true;
+        Bytes.set res !offset x;
+        incr offset
+    | _ as x ->
+        if !upper then (
+          Bytes.set res !offset (Char.uppercase_ascii x);
+          upper := false
+        ) else
+          Bytes.set res !offset x;
+        incr offset in
+  String.iter f s;
+  Bytes.to_string (Bytes.sub res 0 !offset)
+
+(* Translate type names into idiomatic Scala names.  We special case
+ * `string', `int', `bool' and `float` (see code).  For the remainder, we remove
+ * underscores and capitalise any character that is immediately following
+ * an underscore or digit.  We also capitalise the initial character
+ * e.g. "foo_bar42baz" becomes "FooBar42Baz". *)
+let to_class_name str =
+  match str with
+    | "string" -> "std::string"
+    | "int"    -> "int32_t"
+    | "bool"   -> "boolean"
+    | "float"  -> "float"
+    | _ -> to_camel_case str
+
+(* Per https://cpp-lang.org/files/archive/spec/2.12/01-lexical-syntax.html *)
+let cpp_keywords = [
+  "abstract";
+  "case";
+  "catch";
+  "class";
+  "def";
+  "do";
+  "else";
+  "extends";
+  "false";
+  "final";
+  "finally";
+  "for";
+  "forSome";
+  "if";
+  "implicit";
+  "import";
+  "lazy";
+  "macro";
+  "match";
+  "new";
+  "null";
+  "object";
+  "override";
+  "package";
+  "private";
+  "protected";
+  "return";
+  "sealed";
+  "super";
+  "this";
+  "throw";
+  "trait";
+  "try";
+  "true";
+  "type";
+  "val";
+  "var";
+  "while";
+  "with";
+  "yield";
+]
+
+let is_cpp_keyword =
+  let tbl = Hashtbl.create 200 in
+  List.iter (fun k -> Hashtbl.add tbl k ()) cpp_keywords;
+  fun k -> Hashtbl.mem tbl k
+
+(*
+   Automatically append an underscore to a field name if it is a Scala keyword.
```

Comment:
```suggestion
   Automatically append an underscore to a field name if it is a C++ keyword.
```

Reply:
resolved, also updated list of keywords for C++